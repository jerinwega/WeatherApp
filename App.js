import React, { useState, createContext } from 'react';
import { SafeAreaView, StatusBar } from 'react-native';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import store from './src/redux/store';
import { GluestackUIProvider } from '@/src/components/ui/gluestack-ui-provider';
import AppNavigation from './src/navigation/AppNavigation';
import '@/global.css'; // Assuming Tailwind/nativewind is configured

// Theme context
export const ThemeContext = createContext({
  colorMode: 'light',
  toggleColorMode: () => {},
});

export default function App() {
  const [colorMode, setColorMode] = useState('light');

  const toggleColorMode = () => {
    setColorMode((prev) => (prev === 'light' ? 'dark' : 'light'));
  };

  const bgColor = colorMode === 'light' ? '#FFFFFF' : '#262626';

  return (
    <>
      {/* Top Safe Area + Status Bar */}
      <SafeAreaView style={{ backgroundColor: bgColor }} />
      <StatusBar barStyle={colorMode === 'light' ? 'dark-content' : 'light-content'} />

      {/* Theme + Gluestack + Redux Providers */}
      <ThemeContext.Provider value={{ colorMode, toggleColorMode }}>
        <Provider store={store}>
          <GluestackUIProvider mode={colorMode}>
              <NavigationContainer>
                <AppNavigation />
              </NavigationContainer>
          </GluestackUIProvider>
        </Provider>
      </ThemeContext.Provider>
    </>
  );
}
