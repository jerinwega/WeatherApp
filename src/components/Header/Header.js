import React, { useContext } from 'react';
import { HStack, Text, Button, ButtonIcon } from '@/src/components/ui';
import { ThemeContext } from '../../../App';
import { Moon, Sun } from 'lucide-react-native';
import { WHITE_COLOR, BLACK_COLOR } from '../../utils/constants';

const Header = () => {
  const { colorMode, toggleColorMode } = useContext(ThemeContext);

  return (
      <HStack
        className="justify-between items-center px-2 py-3" space="md"
      >
        <Text numberOfLines={1} size="2xl" italic className="flex-1 text-blue-600 font-black">
          NET WORLD SPORTS
        </Text>

        <Button
          variant="link"
          onPress={toggleColorMode}
          className={`p-3 rounded-full ${colorMode === 'dark' ? 'bg-white' : 'bg-black'}`}
        >
          <ButtonIcon
            as={colorMode === 'light' ? Moon : Sun}
            size='xl'
            color={colorMode === "light" ? WHITE_COLOR : BLACK_COLOR}
          />
        </Button>
      </HStack>
  );
};

export default Header;
