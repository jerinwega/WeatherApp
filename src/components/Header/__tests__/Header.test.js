import React from 'react';
import { render } from '@testing-library/react-native';
import Header from '../Header';
import { ThemeContext } from '../../../../App';

jest.mock('expo-constants', () => ({
  expoConfig: {
    extra: {
      OPENWEATHER_API_KEY: 'mock-api-key',
    },
  },
}));

describe('Header Component', () => {
  const mockToggle = jest.fn();


  it('matches snapshot in light mode', () => {
    const { toJSON } = render(
      <ThemeContext.Provider value={{ colorMode: 'light', toggleColorMode: mockToggle }}>
        <Header />
      </ThemeContext.Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

});
