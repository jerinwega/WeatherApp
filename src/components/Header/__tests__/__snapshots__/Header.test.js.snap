// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot in light mode 1`] = `
<View
  className="flex-row gap-3 justify-between items-center px-2 py-3"
>
  <Text
    className="font-body text-2xl italic flex-1 text-blue-600 font-black"
    numberOfLines={1}
  >
    NET WORLD SPORTS Â»
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="group/button flex-row items-center justify-center data-[focus-visible=true]:web:outline-none data-[focus-visible=true]:web:ring-2 data-[disabled=true]:opacity-40 gap-2 border-primary-300 data-[hover=true]:border-primary-400 data-[active=true]:border-primary-500 data-[focus-visible=true]:web:ring-indicator-info h-10 data-[hover=true]:bg-transparent data-[active=true]:bg-transparent p-3 rounded-full bg-black"
    collapsable={false}
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      role="img"
      states={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      stroke="#FFFFFF"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
